runtimes/eden/src/xcm_config.rs

```diff
@@ -151,3 +162,76 @@ impl cumulus_pallet_xcm::Config for Runtime {
 	type RuntimeEvent = RuntimeEvent;
 	type XcmExecutor = XcmExecutor<XcmConfig>;
 }
+pub struct AccountIdToMultiLocation;
+impl Convert<AccountId, MultiLocation> for AccountIdToMultiLocation {
+	fn convert(account: AccountId) -> MultiLocation {
+		X1(AccountId32 {
+			network: NetworkId::Any,
+			id: account.into(),
+		})
+		.into()
+	}
+}
+parameter_types! {
+	pub const BaseXcmWeight: XcmWeight = 100_000_000; // TODO: recheck this
+	pub const MaxAssetsForTransfer: usize = 2;
+}
+parameter_types! {
+	pub SelfLocation: MultiLocation = MultiLocation::here();
+}
+parameter_type_with_key! {
+	pub ParachainMinFee: |_location: MultiLocation| -> Option<u128> {
+		Some(u128::MAX)
+	};
+}
+#[derive(Encode, Decode, Eq, PartialEq, Clone, PartialOrd, Ord, TypeInfo, RuntimeDebug)]
+pub enum CurrencyId {
+	// NODL native token
+	NodleNative,
+}
+pub struct CurrencyIdConvert;
+impl Convert<CurrencyId, Option<MultiLocation>> for CurrencyIdConvert {
+	fn convert(id: CurrencyId) -> Option<MultiLocation> {
+		match id {
+			CurrencyId::NodleNative => Some(NodlLocation::get()),
+		}
+	}
+}
+impl Convert<MultiLocation, Option<CurrencyId>> for CurrencyIdConvert {
+	fn convert(location: MultiLocation) -> Option<CurrencyId> {
+		if location == MultiLocation::parent() {
+			return None;
+		}
+		match location.interior() {
+			&Junctions::X2(Parachain(para_id), PalletInstance(key)) => {
+				if (
+					u32::from(ParachainInfo::parachain_id()),
+					<Balances as PalletInfoAccess>::index() as u8,
+				) == (para_id, key)
+				{
+					Some(CurrencyId::NodleNative)
+				} else {
+					None
+				}
+			}
+			_ => None,
+		}
+	}
+}
+
+impl orml_xtokens::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Balance = Balance;
+	type CurrencyId = CurrencyId;
+	type CurrencyIdConvert = CurrencyIdConvert;
+	type AccountIdToMultiLocation = AccountIdToMultiLocation;
+	type SelfLocation = SelfLocation;
```

Comment:
Can you test if `NodlLocation` can work as `SelfLocation` here? I know it would compile and if it works it's a more consistent way of addressing `here`. But I'm not sure if it can change any behaviour in an actual xtoken call.

Reply:
Having the following error when replacing Selflocation by NodlLocation 
`Failed execute transfer message with Unroutable    `

Reply:
More reading [here](https://github.com/open-web3-stack/open-runtime-module-library/blob/master/xtokens/README.md): 
`Notice the implementation for now also relies on SelfLocation which is already in xtokens config. The SelfLocation can be set to the absolute view (1, Parachain(THIS_PARACHAIN_ID)) and refers to the sender parachain. The alternative is to set SelfLocation to relative view (0, Here) to adhere to Polkadot guidelines.

We use SelfLocation to fund fee to sender's parachain sovereign account on destination parachain, which asset is originated from sender account on sender parachain. This means if user setup too much fee, the fee will not returned to user, instead deposit to sibling parachain sovereign account on destination parachain.

`